src/Tribe/Utils/Theme_Compatibility.php

```diff
@@ -0,0 +1,266 @@
+		}
+
+		$required = in_array( $theme, static::get_registered_themes() );
+
+		return tribe_is_truthy( apply_filters( 'tribe_compatibility_required', $required, $theme ) );
+	}
+
+	/**
+	 * Contains the logic for if this object's classes should be added to the queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $add   Whether to add the class to the queue or not.
+	 * @param array   $class The array of compatibility class names to add.
+	 * @param string  $queue The queue we want to get 'admin', 'display', 'all'.
+	 *
+	 * @return boolean Whether compatibility classes should be added or not.
+	 */
+	public static function should_add_compatibility_class_to_queue( $add, $class, $queue ) {
+		if (
+			'admin' === $queue
+			|| ! static::is_compatibility_required()
+		) {
+			return $add;
+		}
+
+		if ( in_array( $class, static::get_compatibility_classes() ) ) {
+			$add = true;
+		}
+
+		/**
+		 * Filters whether we should add a specific class to the queue.
+		 *
+		 * @since TBD
+		 *
+		 * @param boolean $add   Whether to add the class to the queue or not.
+		 * @param array   $class The array of compatibility class names to add.
+		 * @param string  $queue The queue we want to get 'admin', 'display', 'all'.
+		 */
+		return apply_filters( 'tribe_compatibility_add_class', $add, $class, $queue );
+	}
+
+	/**
+	 * Add compatibility classes.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public static function add_compatibility_classes() {
+		tribe( Compatibility_Classes::class )->add_classes( static::get_compatibility_classes() );
+	}
+
+	/**
+	 * Fetches the correct class strings for theme and child theme if available + the container class.
+	 *
+	 * @since TBD
+	 *
+	 * @return array $classes
+	 */
+	public static function get_container_classes() {
+		$classes =  [ 'tribe-compatibility-container' ];
+
+		if ( static::is_compatibility_required() ) {
+			$classes = array_merge( $classes, static::get_compatibility_classes() );
+		}
+
+		/**
+		 * Filters the HTML classes applied to a compatibility container.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $html_classes Array of classes used for this container.
+		 */
+		return apply_filters( 'tribe_compatibility_container_classes', $classes );
+	}
+
+	/**
+	 * Fetches the correct class strings for theme and child theme if available.
+	 *
+	 * @since TBD
+	 *
+	 * @return array $classes
+	 */
+	public static function get_compatibility_classes() {
+		$classes      = [];
+		$current_theme = static::get_current_theme( true );
+
+		if ( empty( $current_theme ) || empty( $current_theme->get_template() ) ) {
+			return $classes;
+		}
+
+		// Detect if we're using a child theme.
+		if ( $parent = $current_theme->parent ) {
+			$classes[] = sanitize_html_class( 'tribe-theme-' . $parent->get_template() );
+			$classes[] = sanitize_html_class( 'tribe-theme-child-' . $current_theme->get_template() );
+		} else {
+			$classes[] = sanitize_html_class( 'tribe-theme-' . $current_theme->get_template() );
```

Comment:
Will these be passed down to the `Element_Classes`?

Reply:
They aren't right now.
Compatibility classes go on compatibility containers _only_ so it would be bad form (and a lot of noise) to add them to every element that we output.
From the other side, the container needs no additional classes, so I didn't see a need to add E_C at first, although I can see how the additional functionality might be useful?

Reply:
No I think we are good like that. Just double checking.

Reply:
Adding a second thought: if, for some reason we _want_ to add compatibility classes (like body classes in the interim for backwards compatibility) we should do it manually at the point we'd call Element_Classes - easier to clean up?